### ISSUE TITLE
Range validation when the end passes EOF

### ISSUE THREAD

**Issue by jfaust** [MarcusGrass/http-range-header#2]
I'll preface this by saying I'm not sure if this is an issue in this crate or not.

I ran into an issue with `tower-http` (filed at https://github.com/tower-rs/tower-http/issues/384), where LinkedIn's scraper bot fails to retrieve a site due to it passing a `range: bytes=0-3145727` with every request. This causes `http-range-header::ParsedRanges::validate()` to fail.

Initially I thought web servers were just being lax in allowing this, but in reading the RFC (https://www.ietf.org/rfc/rfc2616.txt), I found:
```
10.4.17 416 Requested Range Not Satisfiable

   A server SHOULD return a response with this status code if a request
   included a Range request-header field (section 14.35), and none of
   the range-specifier values in this field overlap the current extent
   of the selected resource, and the request did not include an If-Range
   request-header field. (For byte-ranges, this means that the first-
   byte-pos of all of the byte-range-spec values were greater than the
   current length of the selected resource.)

   When this status code is returned for a byte-range request, the
   response SHOULD include a Content-Range entity-header field
   specifying the current length of the selected resource (see section
   14.16). This response MUST NOT use the multipart/byteranges content-
   type.
```

`none of the range-specifier values in this field overlap` makes me think that `validate()` should not care if the end of the range is past EOF, so long as the start of the range is valid. I believe this effectively allows `LinkedInBot` to make a single request that is guaranteed not to return more than ~3MB, without needing to first do a `HEAD` request, and I'm wondering if that was the intent in the RFC. I'm definitely out of my depth here though, and have not read the entire RFC.

So... I'm not sure if this is a situation where `http-range-header` has the correct behavior, but web servers have decided it's OK to be lax, or if the validation in this case is incorrect.

Thoughts?

**Comment by jfaust** [comment]
Looks like this is actually specified behavior in the spec, from section `14.35.1`:
```
If the last-byte-pos value is absent, or if the value is greater than
   or equal to the current length of the entity-body, last-byte-pos is
   taken to be equal to one less than the current length of the entity-
   body in bytes.
```

**Comment by MarcusGrass** [comment]
Closed by https://github.com/MarcusGrass/http-range-header/pull/3, thanks a lot!

**Comment by MarcusGrass** [comment]
Now published as `0.3.1`
